### DISCUSSION TITLE
Recipient address rejected: User unknown in virtual mailbox table

### DISCUSSION THREAD

**Discussion by kale1d0code** [Mailu/Mailu#3482] [User support]
## Environment & Version

- `docker compose version` Docker Compose version v2.21.0
- Version: `master`

2.0

If you are not using docker compose do not file any new issue here.
Kubernetes related issues belong to https://github.com/Mailu/helm-charts/issues
If you are not using docker compose or kubernetes, create a new thread on user support in [disussions](https://github.com/Mailu/Mailu/discussions/categories/user-support).
Non-bug reports (or bug reports that do not follow the template) are moved to [disussions](https://github.com/Mailu/Mailu/discussions).

## Description

User accounts must exist within mailu for transport rules to kick in
This makes it difficult to route messages to email addresses not stored within mailu,

currently to mitigate this issue, the user accounts that exist in another mail service have to be created in mailu also so that the "User unknown in virtual mailbox table" error does not trigger and the transport rules can be executed.

## Replication Steps

create transport rules to send certain address to another mail service.
attempt to send a message to an address that exists on this other mail service from an address in mailu

## Observed behaviour

message is not transported to the other mail service and the following error message is printed within the smtp service instead
"Recipient address rejected: User unknown in virtual mailbox table"

## Expected behaviour

message to be transported to the other mail service for delivery and for the "Recipient address rejected: User unknown in virtual mailbox table" error to not be triggered.

**Comment by murphysprings** [comment]
I am experiencing the same issue. Thanks for providing a workaround.
MailU 2024.06
I will create an issue; I did see that #3481 was created and closed, but this is still an issue with an inelegant workaround.

**Reply by nextgens** [reply]
Knowingly reopening issues that have been closed will get you banned.

If this is a feature request spec it out properly and put it in the adequate category. As it stands it is unclear why you think that mailu should facilitate overriding 'just' the transport maps. If you are overriding that you can override the rest too, including the virtual_mailboxes or virtual_alias_maps as required.

**Comment by murphysprings** [comment]
The solution that seems to work for me is overriding the `virtual_mailbox_maps` as well as `transport_maps`. Can use the same texthash file for both override directives.

`transport_maps = texthash:/overrides/transport, lmdb:/etc/postfix/transport.map, socketmap:unix:/tmp/podop.socket:transport
virtual_mailbox_maps = texthash:/overrides/transport, socketmap:unix:/tmp/podop.socket:mailbox`
